### DISCUSSION TITLE
OMNeT++ OSPFv2 Pathfinding Issue After Router Failure

### DISCUSSION THREAD

**Discussion by Przemek088** [omnetpp/omnetpp#1245] [Q&A]
Hi,
I have a problem with ospfv2 with inet from example files - dynamic test. Namely, the problem is that when I run a shutdown of the router e.g. R5 at 60 seconds to 90 seconds, OSPF doesn't find another alternate path and doesn't ping with it, it just sort of doesn't see that R5 is still shutdown. Generally it still wants to route packets through the shutdown router and I don't know why, because, for example, RIP immediately detects the failure and looks for another path, and this OSPF doesn't even do reconvergence after the failure.

![obraz_2024-10-23_162329589](https://github.com/user-attachments/assets/814833dc-430f-43af-a677-c3fc3bd44805)

**Comment by avarga** [comment]
We need to reproduce your simulation in order to be able to help. However, the network in the `dynamictest` OSPFv2 example in INET looks quite different from the one you posted. Can you attach your NED, INI, OSPF config and other files so we can try?

**Reply by Przemek088** [reply]
![obraz_2024-10-24_170424730](https://github.com/user-attachments/assets/76bbdc4a-e67e-4235-ae1b-7d7f9d70ed57)
interestingly, the network detects the failure and converges, albeit 30 seconds after its occurrence, i.e. the failure occurs at t=60s and its detection and convergence occurs at t=90s

**Comment by avarga** [comment]
OSPF detects topology changes (e.g. router down) by the absence of regular Hello messages from the neighbor. Are helloInterval and routerDeadInterval set to the default 10s and 40s in your configuration? Because if routerDeadInterval is 40s, then it takes 40s for the neighbors to detect that R5 is down, so 30s (from t=60s till t=90s) is not enough for OSPF to react.

**Reply by Przemek088** [reply]
[ospfv2_ned.txt](https://github.com/user-attachments/files/17508652/ospfv2_ned.txt)
[ospfv2_ini.txt](https://github.com/user-attachments/files/17508653/ospfv2_ini.txt)
[ospfv2_asconfig.txt](https://github.com/user-attachments/files/17508654/ospfv2_asconfig.txt)
[ospfv2_scenario_failure.txt](https://github.com/user-attachments/files/17508655/ospfv2_scenario_failure.txt)
This is all configuration of my network. Everything here until the crash pings beautifully, but after the crash it takes 30 seconds to detect the new path and that the R5 router has died, which is very strange.

**Comment by Przemek088** [comment]
Okey, i have solved this problem. The problem Was XML. When i change code from previous to this:
![obraz_2024-10-24_195240637](https://github.com/user-attachments/assets/e06a8a61-d5f9-4eb3-86eb-c9aa21676c6c)
convergence time change to 2 seconds and failure was succesfully detected.

**Comment by avarga** [comment]
If I get it correctly, the change was to modify helloInterval and routerDeadInterval from the default 10s and 40s that I mentioned above, to 1s and 3s?
